File: c2py_Ver00.03_00/c2py.py
import sys # sys.stdout.write

def c2py_str(rhs, i):
    lhs = rhs[ i[0] ]
    i[0] += 1
    return lhs

def c2py_int(rhs, i):
    lhs = int(rhs[ i[0] ])
    i[0] += 1
    return lhs

def c2py_double(rhs, i):
    lhs = float(rhs[ i[0] ])
    i[0] += 1
    return lhs

def c2py_vecInt(rhs, i):
    size = int(rhs[ i[0] ])
    i[0] += 1
    
    vec = [int(0) for n in range(size)]
    for n in range(size):
        vec[n] = int(rhs[ i[0] ])
        i[0] += 1
    return vec

def c2py_vecDouble(rhs, i):
    size = int(rhs[ i[0] ])
    i[0] += 1
    
    vec = [0.0 for n in range(size)]
    for n in range(size):
        vec[n] = float(rhs[ i[0] ])
        i[0] += 1
    return vec

def c2py_vvecInt(rhs, i):
    row_size = int(rhs[ i[0] ])
    i[0] += 1

    vvec = [[] for p in range(row_size)]
    
    for p in range(row_size):
        col_size = int(rhs[ i[0] ])
        i[0] += 1
        
        vvec[p] = [int(0) for q in range(col_size)]
        
        for q in range(col_size):
            vvec[p][q] = int(rhs[ i[0] ])
            i[0] += 1
            
    return vvec

def c2py_vvecDouble(rhs, i):
    row_size = int(rhs[ i[0] ])
    i[0] += 1

    vvec = [[] for p in range(row_size)]
    
    for p in range(row_size):
        col_size = int(rhs[ i[0] ])
        i[0] += 1
        
        vvec[p] = [0.0 for q in range(col_size)]
        
        for q in range(col_size):
            vvec[p][q] = float(rhs[ i[0] ])
            i[0] += 1
            
    return vvec

def print_vec(vec):
    for q in range(len(vec)):
        sys.stdout.write("%8.3lf " % vec[q])
    sys.stdout.write("\n")

def print_vvec(vvec):
    for p in range(len(vvec)):
        for q in range(len(vvec[p])):
            sys.stdout.write("%8.3lf " % vvec[p][q])
        sys.stdout.write("\n")

File: c2py_Ver00.03_00/c2py.cpp
#include "c2py.hpp"

std::string c2py_str(const char* rhs){ return sstd::ssprintf(" %s", rhs); }
std::string c2py_int(int rhs)        { return sstd::ssprintf(" %d", rhs); }
std::string c2py_double(double rhs)  { return sstd::ssprintf(" %lf", rhs); }

std::string c2py_vecInt(std::vector<int>& rhs){

	std::string lhs = sstd::ssprintf(" %d", rhs.size());
	for(uint i=0; i<rhs.size(); i++){
		lhs += sstd::ssprintf(" %d", rhs[i]);
	}
	return lhs;
}
std::string c2py_vecDouble(std::vector<double>& rhs){

	std::string lhs = sstd::ssprintf(" %d", rhs.size());
	for(uint i=0; i<rhs.size(); i++){
		lhs += sstd::ssprintf(" %lf", rhs[i]);
	}
	return lhs;
}

std::string c2py_vvecInt(std::vector<std::vector<int>>& rhs){

	std::string lhs = sstd::ssprintf(" %d", rhs.size());
	for(uint i=0; i<rhs.size(); i++){
		lhs += sstd::ssprintf(" %d", rhs[i].size());
		for(uint i=0; i<rhs[i].size(); i++){
			lhs += sstd::ssprintf(" %d", rhs[i]);
		}
	}
	return lhs;
}
std::string c2py_vvecDouble(std::vector<std::vector<double>>& rhs){

	std::string lhs = sstd::ssprintf(" %d", rhs.size());
	for(uint p=0; p<rhs.size(); p++){
		lhs += sstd::ssprintf(" %d", rhs[p].size());
		for(uint q=0; q<rhs[p].size(); q++){
			lhs += sstd::ssprintf(" %lf", rhs[p][q]);
		}
	}
	return lhs;
}
